pkg/types/intoto/intoto_schema.json

```diff
@@ -6,7 +6,7 @@
     "type": "object",
     "oneOf": [
         {
-            "$ref": "v0.0.1/intoto_v0_0_1_schema.json"
+            "$ref": "v0.0.2/intoto_v0_0_2_schema.json"
```

Comment:
Need some help on writing up the test for this. I see the script `issue-872-e2e-test.sh` to test this but not sure what needs to be done to get a e2e test for it.

Reply:
I think the existing harness tests might cover this? Older versions will insert intoto attestations at v001, and later when we test while running @ HEAD we can ensure that they can still be retrieved.

We would just need to update the harness tests to test against the 3 previous versions AND main (we don't test against main at the moment).

Reply:
i tested locally and you can still upload 0.0.1 intoto entries in to the log. Once you rebase ontop of #1003 and #1004, I think we should add the following code to `pkg/types/intoto/intoto.go`:
```
// SupportedVersions returns the supported versions for this type;
// it deliberately omits 0.0.1 from the list of supported versions as that
// version did not persist signatures inside the log entry
func (it BaseIntotoType) SupportedVersions() []string {
	return []string{"0.0.2"}
}
```
We may have to figure out how to adapt the unit tests accordingly as I believe the `TestIntoto` is used by the harness and issue872 tests, but will probably fail once you do the rebase.

Reply:
also I think we should have both of them listed here (otherwise the codegen becomes disconnected from the source, even though you're technically not deleting it in the PR)

Reply:
Rebased to include the new PRs and added `SupportedVersions`. The harness tests do fail when set to `intotoModel := &models.IntotoV001Schema{}. Currently setting them to `intotoModel := &models.IntotoV002Schema{}`.